### DISCUSSION TITLE
mass fraction at PorousFlowPeacemanBorehole

### DISCUSSION THREAD

**Discussion by lynnmunday** [idaholab/moose#20589] [Q&A General] [answered]
I have a single phase, 2 component simulation (variables are massFracCold , massFracHot, Pore_Pressure).  Injection is with massFracCold using a PorousFlowPointSourceFromPostprocessor and production is with a PorousFlowPeacemanBorehole.  When I look at the exodus output for the massFraction for each phase, they vary between 0 and 1 everywhere in the domain except at the peaceman borehole.  At the peaceman borehole, which coincides with a node, the massfraction for the injected phase (massFracCold) just keeps climbing and the initial phase (massFracHot) that is being displaced levels off.  It's like the values at the production node are being accumulated.  Is this the expected behavior at a production point?   

Here are what I think are the important parts of my input file (Some of these are falcon objects in a PR that has not been merged yet):
```
[Variables]
  [Pore_Pressure]
  []
  [massFracCold]
    initial_condition = 0.00001
  []
  [massFracHot]
    initial_condition = 0.99999
  []
[]

[PorousFlowFullySaturated]
  porepressure = Pore_Pressure
  mass_fraction_vars = 'massFracCold massFracHot'
  coupling_type = Hydro
  fp = simple_fluid
  stabilization = Full
[]

[UserObjects]
  [kg_out_uo]
    type = PorousFlowSumQuantity
  []
[]

[Modules]
  [FluidProperties]
    [simple_fluid]
      type = SimpleFluidProperties
      viscosity = 0.0001
    []
  []
[]

[DiracKernels]
  [inject_fluid_mass]
    type = PorousFlowReporterPointSourcePP
    mass_flux = inject_mass_flux
    x_coord_reporter = 'inject_node/node_x'
    y_coord_reporter = 'inject_node/node_y'
    z_coord_reporter = 'inject_node/node_z'
    variable = massFracCold
  []
  [withdraw_fluid]
    type = PorousFlowPeacemanBorehole
    fluid_phase = 0
    SumQuantityUO = kg_out_uo
    bottom_p_or_t = insitu_pp_borehole
    character = 1
    line_length = 1
    x_coord_reporter = 'prod_node/node_x'
    y_coord_reporter = 'prod_node/node_y'
    z_coord_reporter = 'prod_node/node_z'
    weight_reporter = 'prod_node/node_value'
    unit_weight = '0 0 0'
    use_mobility = true
    variable = Pore_Pressure
  []
[]
```

**Comment by lynnmunday** [comment]
@cpgr or @WilkAndy

**Comment by cpgr** [comment, marked as answer]
So is massfrachot being withdrawn and massfraccold not? You might need another PeacemanBorehole for the massfraccold variable and use the `mass_fraction_component` input parameter to specify the cold mass fraction? 

Andy would probably know better than me so hopefully he can chime in!

**Comment by WilkAndy** [comment]
Are you sure you have 2 components?  I think you have 3 components: MassFracCold, MassFracHot and 1 - (MassFracCol + MassFracHot).

You will have to know the order of the unknowns: what Variable is associated with which fluid-component equation.  That's defined at the top of https://mooseframework.inl.gov/source/actions/PorousFlowFullySaturated.html

**Reply by WilkAndy** [reply]
Sorry for being terse - i'm in a day-long workshop :-)

**Comment by lynnmunday** [comment]
@WilkAndy I didn't read your tutorial close enough!  Thanks for pointing me to the documentation for that action.  I did have three components.  I fixed that and put in another peaceman borehole for the new fluid like @cpgr pointed out and used the mass_fraction and it all seems to be working.
